app/uk/gov/hmrc/eusubsidycompliancestub/controllers/UndertakingController.scala

```diff
@@ -150,7 +150,18 @@ class UndertakingController @Inject()(
     eori match {
       case a if a.endsWith("999") => // fake 500
         Future.successful(InternalServerError(Json.toJson(errorDetailFor500)))
-      case b if b.endsWith("888") || Store.undertakings.retrieveByEori(eori).isEmpty => // fake not found (ideally should have been 404)
+
+      case b if b.endsWith("777")  => // ID invalid
+        val noUndertakingFoundResponse: JsValue = Json.obj(
+          "retrieveUndertakingResponse" -> Json.obj(
+            "responseCommon" -> badResponseCommon(
+              "055",
+              "ID number missing or invalid"
```

Comment:
I don't think invalid EORIs should be able to get to here. Is this just for "EORI doesn't exist"?

Reply:
I have just added a scenario here like if EORI ends with 777, then I should get this error code, so that I can test it on FE